server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -294,15 +295,19 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 builder.field("remote", remoteInfo);
             }
             builder.array("index", getSearchRequest().indices());
-            builder.array("type", getSearchRequest().types());
+            String[] types = getSearchRequest().types();
+            if (types != null && types.length > 0) {
```

Comment:
Super small comment -- I don't think these null checks are necessary, since `SearchRequest` validates that the `types` array can't be null. Similarly, `IndexRequest` validates that the type is not null.

Reply:
> `IndexRequest` validates that the type is not null

Unfortunately this is done in `validate()` only, so its possible to set a `null` type via the ctors or the setter and then xContent serialization would break. For `SearchRequest` I think you are right and types cannot be `null` so I will remove the check there but leave it for IndexRequest.